### PR TITLE
3/no browser

### PR DESCRIPTION
**Pull request by abea**
Deprecates the `browser` module option and related method, `getComponentName`.

### CODE REVIEW

File: modules/@apostrophecms/module/index.js [hunk 1]
```diff
@@ -509,14 +507,6 @@ module.exports = {
         return _.get(self.options, dotPathOrArray, def);
       },
 
-      // If `name` is `manager` and `options.browser.components.manager` is set,
-      // return that string, otherwise return `def`. This is used to decide what
-      // Vue component to instantiate on the browser side.
-
-      getComponentName(name, def) {
-        return _.get(self.options, `browser.components.${name}`, def);
```
Comment (abea) [inline modules/@apostrophecms/module/index.js:517]: An alternative would be to update `getComponentName` to use `options.components` instead. No modules are currently using that option, but some, like the modal module, do open the door:

```
  getBrowserData(req) {
    return {
      modals: self.modals,
      components: {
        the: options.components.the || 'TheAposModals',
        confirm: options.components.confirm || 'AposModalConfirm'
      }
    };
  }
```

Comment (abea) [reply modules/@apostrophecms/module/index.js:517]: ... I'll do that.

File: modules/@apostrophecms/module/index.js [hunk 2]
```diff
@@ -509,12 +507,11 @@ module.exports = {
...
       },
 
-      // If `name` is `manager` and `options.browser.components.manager` is set,
+      // If `name` is `manager` and `options.components.manager` is set,
       // return that string, otherwise return `def`. This is used to decide what
       // Vue component to instantiate on the browser side.
-
       getComponentName(name, def) {
-        return _.get(self.options, `browser.components.${name}`, def);
+        return _.get(self.options, `components.${name}`, def);
```
Comment (boutell) [inline modules/@apostrophecms/module/index.js:514]: This is the same ol' problem: you have to reset the whole thing, including new components you might not know about that are added to that module's configurable components object later, just to change one. It's askin' for trouble.

So I would recommend:

```
return self.options[`${name}Component`] || def
```

Which can be overridden with no surprises.

File: modules/@apostrophecms/rich-text-widget/index.js [hunk 3]
```diff
@@ -324,12 +250,88 @@ module.exports = {
       },
       // Add on the core default options to use, if needed.
       getBrowserData(_super, req) {
-        const result = _super(req);
+        const initialData = _super(req);
 
-        _.defaults(result, {
+        const rteData = {
+          components: {
```
Comment (boutell) [inline modules/@apostrophecms/rich-text-widget/index.js:256]: If we go with the `${name}Component` pattern we can set the original defaults as options which is more self-documenting and can potentially be extracted for documentation.

File: modules/@apostrophecms/rich-text-widget/index.js [hunk 4]
```diff
@@ -324,12 +250,88 @@ module.exports = {
...
+            widgetEditor: 'AposRichTextWidgetEditor',
+            widget: 'AposRichTextWidget'
+          },
+          tools: {
```
Comment (boutell) [inline modules/@apostrophecms/rich-text-widget/index.js:260]: Same comment for tools.

**Review by boutell — CHANGES_REQUESTED** [review]

File: modules/@apostrophecms/module/index.js (hunk 2 above)
Comment (abea) [reply modules/@apostrophecms/module/index.js:514]: That's a new bucket of worms: creating a component name structure. Maybe we should actually remove this for now and create a ticket to come up with a better way for project-level component override.

Comment (boutell) [reply modules/@apostrophecms/module/index.js:514]: Rich text widgets already need and use it, so we can't punt. We could discuss in a meeting.

Comment (abea) [reply modules/@apostrophecms/module/index.js:514]: Rich text widgets don't use this method currently. It hasn't really been doing anything lately, I don't think. Not other than returning the default values we pass in.

Comment (abea) [reply modules/@apostrophecms/module/index.js:514]: The goal of this PR isn't to establish a new and better way to set module components. It's to get rid of the barely- and inconsistently-used `browser` option. If we switch to `components` option instead, for now, we've not gained or lost anything there other than not having to dig deeper in a dot path.

File: modules/@apostrophecms/rich-text-widget/index.js (hunk 3 above)
Comment (abea) [reply modules/@apostrophecms/rich-text-widget/index.js:256]: [See above](https://github.com/apostrophecms/apostrophe/pull/2766/files?file-filters%5B%5D=.js#r584841341). I don't think we should make this call in this PR. I can return these to the options object in a `components` option.

File: modules/@apostrophecms/rich-text-widget/index.js (hunk 4 above)
Comment (abea) [reply modules/@apostrophecms/rich-text-widget/index.js:260]: As above, I can move this back into the options in its own `tools` option.

**Review by boutell — APPROVED** [review]
That works for now.

**Pull request merged**